File: CS/FolderBrowserDialogServiceSample/ViewModels/FolderBrowserDialogViewModel.cs
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;

namespace FolderBrowserDialogServiceSample.ViewModels
{
    [POCOViewModel]
    public class FolderBrowserDialogViewModel {
        public virtual string Description { get; set; }
        public virtual bool ShowNewFolderButton { get; set; }
        public virtual string ResultPath { get; set; }
        protected virtual IFolderBrowserDialogService FolderBrowserDialogService { get { return this.GetService<IFolderBrowserDialogService>(); } }

        public FolderBrowserDialogViewModel() {
            ShowNewFolderButton = true;
        }
        public void ShowDialog() {
            if (FolderBrowserDialogService.ShowDialog())
                ResultPath = FolderBrowserDialogService.ResultPath;
        }
    }
}

File: VB/FolderBrowserDialogServiceSample/ViewModels/FolderBrowserDialogViewModel.vb
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO

Namespace FolderBrowserDialogServiceSample.ViewModels

    <POCOViewModel>
    Public Class FolderBrowserDialogViewModel

        Public Overridable Property Description As String

        Public Overridable Property ShowNewFolderButton As Boolean

        Public Overridable Property ResultPath As String

        Protected Overridable ReadOnly Property FolderBrowserDialogService As IFolderBrowserDialogService
            Get
                Return GetService(Of IFolderBrowserDialogService)()
            End Get
        End Property

        Public Sub New()
            ShowNewFolderButton = True
        End Sub

        Public Sub ShowDialog()
            If FolderBrowserDialogService.ShowDialog() Then ResultPath = FolderBrowserDialogService.ResultPath
        End Sub
    End Class
End Namespace
